lib/cell/collection.rb

```diff
@@ -2,25 +2,19 @@ module Cell
   class Collection
     def initialize(ary, options, cell_class)
       options.delete(:collection)
-      @method     = options.delete(:method)           # TODO: remove in 5.0.
-      @join       = options.delete(:collection_join)  # TODO: remove in 5.0.
+      self.method = options.delete(:method)                   if options.include?(:method)          # TODO: remove in 5.0.
```

Comment:
Note that these if statements replace the `deprecate_options!` method call below. In the end, less code will be executed when deprecated options are not used, and about the same amount of code will be executed when deprecated options are used.